### PR TITLE
Add --ner-label-index option to validate

### PR DESCRIPTION
**Pull request by claire-yq**
Modifies the `validate` method by adding an optional `--ner-label-index` flag, which specifies the column for the NER label. The default value is -1.

### CODE REVIEW

File: seqscore/conll.py [hunk 1]
```diff
@@ -73,8 +73,8 @@ def from_line(cls, line: str, line_num: int, source_name: str) -> "_CoNLLToken":
                 )
 
         text = splits[0]
-        label = splits[-1]
-        other_fields = tuple(splits[1:-1])
+        label = splits[ner_label_index]
+        other_fields = tuple(splits[1:ner_label_index] + splits[ner_label_index + 1:])
```
Comment (ConstantineLignos) [inline seqscore/conll.py]: This is unfortunately going to mean we need a larger change and might need to abandon the current behavior for other_fields. Let's go over it tomorrow.

File: tests/test_validation_click.py [hunk 2]
```diff
@@ -52,6 +52,16 @@ def test_valid_bio_twofiles() -> None:
     assert result.exit_code == 0
 
 
+def test_valid_bio_ner_label_index() -> None:
+    runner = CliRunner()
+    result = runner.invoke(
+        validate,
+        ["--labels", "BIO", "--ner-label-index", 1, os.path.join("tests", "conll_annotation", "labels_not_last_col.bio")],
```
Comment (ConstantineLignos) [inline tests/test_validation_click.py]: Add another test with the index as -2 instead so we can test positive and negative indices.

**Review by ConstantineLignos — COMMENTED** [review]
Some quick comments to look at before we meet tomorrow. Can you check what is causing the test failures as well?

File: seqscore/conll.py (hunk 1 above)
Comment (claire-yq) [reply seqscore/conll.py]: The logic here is what was causing the test failures (when `ner_label_index = -1`, `ner_label_index + 1 = 0`, so all the fields get copied into `other_fields`). I fixed that issue, but I'm guessing you're talking about the other issue where the order of `other_fields' might be incorrect?

Comment (ConstantineLignos) [reply seqscore/conll.py]: Yes, I think the original logic is going to need to be updated since it makes a lot of assumptions.

**Comment by ConstantineLignos** [comment]
We’ll meet this afternoon, but before then please sort out why tests only pass on some Python versions and make sure to explicitly block the NER label index being zero (that is reserved for the token at the moment, but we’ll revisit).
